yt_dlp/extractor/lbry.py

```diff
@@ -197,23 +224,26 @@ def _real_extract(self, url):
         display_id = compat_urllib_parse_unquote(display_id)
         uri = 'lbry://' + display_id
         result = self._resolve_url(uri, display_id, 'stream')
+        headers = {'Referer': 'https://odysee.com/'}
         if result['value'].get('stream_type') in self._SUPPORTED_STREAM_TYPES:
-            claim_id, is_live, headers = result['claim_id'], False, {}
+            claim_id, is_live = result['claim_id'], False
             streaming_url = self._call_api_proxy(
                 'get', claim_id, {'uri': uri}, 'streaming url')['streaming_url']
             final_url = self._request_webpage(
                 HEADRequest(streaming_url), display_id,
+                headers=headers,
                 note='Downloading streaming redirect url info').geturl()
         elif result.get('value_type') == 'stream':
             claim_id, is_live = result['signing_channel']['claim_id'], True
-            headers = {'referer': 'https://player.odysee.live/'}
             live_data = self._download_json(
                 'https://api.odysee.live/livestream/is_live', claim_id,
                 query={'channel_claim_id': claim_id},
                 note='Downloading livestream JSON metadata')['data']
-            streaming_url = final_url = live_data.get('VideoURL')
-            if not final_url and not live_data.get('Live'):
+            if 'VideoURL' not in live_data or live_data.get('Live') is False:
+                streaming_url = final_url = None
                 self.raise_no_formats('This stream is not live', True, claim_id)
+            else:
+                streaming_url = final_url = live_data.get('VideoURL')
```

Comment:
pls explain this change

Reply:
I had the case, where VideoURL (m3u8) was present, but the video was not live yet. Consequence was, that `--wait-for-video` did not work, because formats were returned anyway.

Reply:
then this should just be `if not live_data.get('Live')`, correct? VideoURL not existing probably indicates a bug instead

Reply:
Yes.